test/test_NameExpander.js

```diff
@@ -0,0 +1,68 @@
+"use strict";
+
+var NameExpander = require("../lib/util/NameExpander");
+var assert = require("assert");
+var path = require("path");
+
+describe("NameExpander", function () {
+  it("should expand to the correct file names", function() {
+    var nameExpander = new NameExpander("test-uri");
+    nameExpander.addLocation(__dirname);
+
+    var expectedFiles = adjustPaths([
+      "test-uri",
+      "test-uri.scss",
+      "test-uri.sass",
+      "test-uri.css",
+      "_test-uri.scss",
+      "_test-uri.sass",
+      "_test-uri.css",
+      "test-uri/index.scss",
+      "test-uri/index.sass",
+      "test-uri/index.css",
+      "test-uri/_index.scss",
+      "test-uri/_index.sass",
+      "test-uri/_index.css"
```

Comment:
The [Ruby Sass importer logic](https://github.com/sass/sass/blob/stable/lib/sass/importers/filesystem.rb#L118-L122) specifies that if an extension is specified, only that extension is considered. So this change is not desired.

It also specifies that if the extension is `.css` then the sass implementation will create a CSS-based import to a relative URI, so our importer should actually never be invoked if the uri being `@import`ed is `foo.css`. If our importer is being called, this is a libsass bug. In Sass 4.0 we're changing to a different directive (`@use`) which will allow for css being invoked like this, so it's ok for our name expander to handle this case, just that we should know it won't actually happen in an integration test.